txn.go

```diff
@@ -663,15 +663,25 @@ func (txn *Txn) getIndexValue(table, index string, args ...interface{}) (*IndexS
 	return indexSchema, val, err
 }
 
-// ResultIterator is used to iterate over a list of results
-// from a Get query on a table.
+// ResultIterator is used to iterate over a list of results from a query on a table.
+//
+// When a ResultIterator is created from a write transaction, the results from
+// Next will reflect a snapshot of the table at the time the ResultIterator is
+// created.
```

Comment:
Should we call out the consistency anomalies more explicitly here?

Perhaps something like:

> Note: this means calling `Insert` or `Delete` on a transaction while iterating is allowed, but those changes will not be observed in the results returned from later calls to `Next`. For example if an item that would appear later in the iterator is deleted it will still be iterated over. If an item is inserted into the index being iterated, it will not be iterated over. A future `Get` will return a new iterator containing all updates however.

Reply:
Sounds good! I added this with a few small edits.